File: decompiled/sources/com/amo/demo/wheelview/ArrayWheelAdapter.java
package com.amo.demo.wheelview;

public class ArrayWheelAdapter<T> implements WheelAdapter {
    public static final int DEFAULT_LENGTH = -1;
    private T[] items;
    private int length;

    public ArrayWheelAdapter(T[] tArr, int i) {
        this.items = tArr;
        this.length = i;
    }

    public ArrayWheelAdapter(T[] tArr) {
        this(tArr, -1);
    }

    public String getItem(int i) {
        if (i < 0) {
            return null;
        }
        T[] tArr = this.items;
        if (i < tArr.length) {
            return tArr[i].toString();
        }
        return null;
    }

    public int getItemsCount() {
        return this.items.length;
    }

    public int getMaximumLength() {
        return this.length;
    }
}

File: decompiled/sources/com/amo/demo/wheelview/ArrayWheelAdapter.kt
package com.amo.demo.wheelview

class ArrayWheelAdapter<T> : WheelAdapter {
    private val items: Array<T>
    private val length: Int

    constructor(items: Array<T>, length: Int) {
        this.items = items
        this.length = length
    }

    constructor(items: Array<T>) : this(items, DEFAULT_LENGTH)

    override fun getItem(i: Int): String? {
        if (i < 0) {
            return null
        }
        if (i < items.size) {
            return items[i].toString()
        }
        return null
    }

    override fun getItemsCount(): Int {
        return items.size
    }

    override fun getMaximumLength(): Int {
        return length
    }

    companion object {
        const val DEFAULT_LENGTH: Int = -1
    }
}
